lib/rules/no-irregular-whitespace/index.cjs

```diff
@@ -65,48 +72,65 @@ const rule = (primary) => {
 		}
 
 		/**
-		 * @param {import('postcss').Node} node
+		 * @template {import('postcss').Node} T
+		 * @param {T} node
 		 * @param {string | undefined} value
+		 * @param {(node: T) => number} getIndex
 		 */
-		const validate = (node, value) => {
-			const issue = value && findIrregularWhitespace(value);
+		const validate = (node, value, getIndex) => {
+			if (!value) return;
+
+			const issues = findIrregularWhitespace(value);
+
+			if (!issues.length) return;
 
-			if (issue) {
+			const startIndex = getIndex(node);
+
+			issues.forEach(({ index, length }) => {
 				report({
 					ruleName,
 					result,
 					message: messages.unexpected,
 					node,
-					word: issue,
+					index: startIndex + index,
+					endIndex: startIndex + index + length,
 				});
-			}
+			});
 		};
 
 		root.walkAtRules((atRule) => {
-			validate(atRule, atRule.name);
-			validate(atRule, atRule.params);
-			validate(atRule, atRule.raws.before);
-			validate(atRule, atRule.raws.after);
-			validate(atRule, atRule.raws.afterName);
-			validate(atRule, atRule.raws.between);
+			validate(atRule, atRule.raws.before, zeroIndex);
+			validate(atRule, atRule.name, oneIndex);
+			validate(atRule, atRule.raws.afterName, nodeFieldIndices.atRuleAfterNameIndex);
+			validate(atRule, getAtRuleParams(atRule), nodeFieldIndices.atRuleParamIndex);
+			validate(atRule, atRule.raws.between, nodeFieldIndices.atRuleBetweenIndex);
+			validate(atRule, atRule.raws.after, nodeFieldIndices.atRuleAfterIndex);
 		});
 
 		root.walkRules((ruleNode) => {
-			validate(ruleNode, ruleNode.selector);
-			validate(ruleNode, ruleNode.raws.before);
-			validate(ruleNode, ruleNode.raws.after);
-			validate(ruleNode, ruleNode.raws.between);
+			validate(ruleNode, ruleNode.raws.before, zeroIndex);
+			validate(ruleNode, getRuleSelector(ruleNode), zeroIndex);
+			validate(ruleNode, ruleNode.raws.between, nodeFieldIndices.ruleBetweenIndex);
+			validate(ruleNode, ruleNode.raws.after, nodeFieldIndices.ruleAfterIndex);
 		});
 
 		root.walkDecls((decl) => {
-			validate(decl, decl.prop);
-			validate(decl, decl.value);
-			validate(decl, decl.raws.before);
-			validate(decl, decl.raws.between);
+			validate(decl, decl.raws.before, zeroIndex);
+			validate(decl, decl.prop, zeroIndex);
+			validate(decl, decl.raws.between, nodeFieldIndices.declarationBetweenIndex);
+			validate(decl, getDeclarationValue(decl), nodeFieldIndices.declarationValueIndex);
 		});
```

Comment:
I suppose `decl.raws.after` doesn't exist.

Reply:
Correct, `after` does not come after the rule/atrule, it is what comes before the closing `}`:

```
.foo { bar: 1;  }
/*            ^^ this is the after of the rule
```

```
.foo { bar: 1  ; }
/*           ^^ this is part of `raws.values.raw`, `1  `
```

Declarations have no closing `}`